pebble/src/main/java/com/mitchellbosecke/pebble/extension/core/DateFilter.java

```diff
@@ -88,19 +103,36 @@ private Object applyDate(Object dateOrString, final PebbleTemplate self, final L
     if (timeZone != null) {
       intendedFormat.setTimeZone(TimeZone.getTimeZone(timeZone));
     }
+
+    if (alterTimeDuration != null) {
+      date = Date.from(date.toInstant().plus(alterTimeDuration));
+    }
+
     return new SafeString(intendedFormat.format(date));
   }
 
-  private Object applyTemporal(final TemporalAccessor input, PebbleTemplate self,
+  private Object applyTemporal(Temporal input, PebbleTemplate self,
       final Locale locale,
-      int lineNumber, final String format, final String timeZone) throws PebbleException {
+      int lineNumber, final String format, final String timeZone,
+      Duration alterTimeDuration) throws PebbleException {
     DateTimeFormatter formatter = format != null
         ? DateTimeFormatter.ofPattern(format, locale)
         : DateTimeFormatter.ISO_DATE_TIME;
 
     ZoneId zoneId = getZoneId(input, timeZone);
     formatter = formatter.withZone(zoneId);
 
+    if (alterTimeDuration != null) {
+      try {
+        input = input.plus(alterTimeDuration);
+      }
+      catch (UnsupportedTemporalTypeException e){
```

Comment:
This needs to be handled better. There are some Java Date types that cannot handle Duration, as it is represented in seconds. For this cases we need Period, which keeps track of years, months, days.

Would it be better to look for all the java types that don't have the time component and check the type here?

Reply:
We might have to. Doing logic as part of an exception flow in imo not a nice thing